06.wc/wc.rb

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'optparse'
+require 'etc'
+WIDTH = 5
+
+def main
+  params = read_option
+  files = ARGV[0].nil? ? [{ name: '', body: readlines.join }] : ARGV.map { |file| { name: file, body: File.open(file).read } }
+  total = files.map { |file| count_and_print_wc_element(params, file) }
+  if files.size >= 2
+    total.transpose.map(&:sum).each { |value| printf("%#{WIDTH}d ", value) }
+    puts('total')
+  end
+  total # rubocop Style/GuardClause: Use a guard claus対策
```

Comment:
rubocop-fjordのチェックを全てパスすることが条件なので、ここもコメントで回避するのではなく解決いただきたいです 🙏

Reply:
すみません、紛らわしコメントを書き、申し訳ありません。🙇
コメントなしでも、rubocopパスします。
<img width="937" alt="スクリーンショット 2022-06-22 13 26 18" src="https://user-images.githubusercontent.com/100653079/174943318-eba55e3d-464b-479b-ac98-ca3c15e09ebc.png">


ここの処理では、コメントで回避しているのではなく、`total`を戻り値にすることで回避しています。
コメントは、「mainの戻り値にわざわざ`total`を指定しているのは、rubocopの指摘を解決するためである」と伝え、
この戻り値に意味がないことを記すものでした。

##
元々のコード
(`Style/GuardClause: Use a guard claus`指摘が出る)
```ruby
def main
  params = read_option
  files = ARGV[0].nil? ? [{ name: '', body: readlines.join }] : ARGV.map { |file| { name: file, body: File.open(file).read } }
  total = files.map { |file| count_and_print_wc_element(params, file) }
  if files.size >= 2
    total.transpose.map(&:sum).each { |value| printf("%#{WIDTH}d ", value) }
    puts('total')
  end
end
```
```ruby
$ rubocop
Inspecting 1 file
C

Offenses:

wc.rb:12:3: C: Style/GuardClause: Use a guard clause (return unless files.size >= 2) instead of wrapping the code inside a conditional expression.
  if files.size >= 2
  ^^

1 file inspected, 1 offense detected
```

##
totalを付けることrubocopの指摘を回避したコード
```ruby
def main
  params = read_option
  files = ARGV[0].nil? ? [{ name: '', body: readlines.join }] : ARGV.map { |file| { name: file, body: File.open(file).read } }
  total = files.map { |file| count_and_print_wc_element(params, file) }
  if files.size >= 2
    total.transpose.map(&:sum).each { |value| printf("%#{WIDTH}d ", value) }
    puts('total')
  end
  total
end
```

Reply:
```ruby
total # rubocopの対策のための戻り値、戻り値がtotalなことに意味はなし
```
こちらのようにコメントを書き換えています。

[Rubyスクリプトにもmainメソッドを定義するといいかも、という話 \- Qiita](https://qiita.com/jnchito/items/4b4cae54170cc2f4377e)
こちらに従い、mainメソッドを定義していますが、戻り値はどのようにしておくのが良いのでしょうか?

C言語なら、`return 0`が関数の正常終了と共通認識がありますが...もしくはVoid型の関数にして戻り値が明示的に無いことが示せる。

Reply:
なるほど、経緯を理解しました。戻り値の話の前に、 `Style/GuardClause` の対処方法が良くないですね。ここで指摘されていることは「ifで処理が終わるのであれば、ifの条件を満たさない場合にアーリーリターンしてネストを深くしないようにしましょう」ということです。

https://www.rubydoc.info/gems/rubocop/0.26.1/RuboCop/Cop/Style/GuardClause

よって、修正もアーリーリターンする方針で対応しましょう。

----

C言語で言うところのmainの戻り値の話ですが、プロセスの終了ステータスを指定したいということですね。今回のプラクティスではそこまでの制御は不要ですが、以下のメソッドを使うことで同様の動作を実現できます。

https://docs.ruby-lang.org/ja/latest/method/Kernel/m/exit.html